test/e2e/spinup_test.go

```diff
@@ -256,13 +256,12 @@ func (s *spinupSuite) Exec(t testing.TB, ctx context.Context, testName string) (
 		}
 	}()
 
-	minioExit := make(chan struct{})
+	var minioExit chan struct{}
 	if s.withPreStartedMinio {
 		// Start minio before anything else.
 		// NewTestBucketFromConfig is responsible for healthchecking by creating a requested bucket in retry loop.
-
 		minioExit, err = newSpinupSuite().
-			Add(minio(s.minioConfig.AccessKey, s.minioConfig.SecretKey), "").
+			Add(minio(ctx, s.minioConfig.AccessKey, s.minioConfig.SecretKey), "").
```

Comment:
ctx not needed

Reply:
Exec will do the same logic already.

Reply:
exec does SIGTERM, ctx deadline do SIGKILL, I'd prefer leave this

Reply:
so let's do SIGKILL in exec (:

Reply:
Right now we have 2 ways of killing BOTH ctx SIGKILL and Process.Signal SIGTERM. This is unnacceptable inconsistency.